cmake/GetMonado_gl.cmake

```diff
@@ -0,0 +1,46 @@
+# module to download, build and install the Monado_gl ILLIXR plugin
+
+# get dependencies
+pkg_check_modules(glslang glslang)
```

Comment:
Not sure if my understanding of my issue is complete, but my CMake build is failing here. It may have to do with my distro - I'm not very familiar with pkg-config, but this issue seems to suggest that certain distros applied their own patches to add support for this: https://github.com/KhronosGroup/glslang/issues/1715.

Running `pkg-config --list-all | grep glslang` also doesn't seem to return anything for me, even though I have `glslang-tools` installed (and verified that I can run `glslang`). As an alternative to this, it seems like [find_package(Vulkan)](https://cmake.org/cmake/help/latest/module/FindVulkan.html) offers some information for utilities like glslang in newer CMake versions, so we could consider requiring a higher version of CMake?

Reply:
To follow up on this, the actual error that I encounter afterwards is this:
```
CMake Error at cmake/GetMonado_gl.cmake:6 (find_package):
  By not providing "Findglslang.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "glslang", but
  CMake did not find one.

  Could not find a package configuration file provided by "glslang"
  (requested version 11) with any of the following names:

    glslangConfig.cmake
    glslang-config.cmake

  Add the installation prefix of "glslang" to CMAKE_PREFIX_PATH or set
  "glslang_DIR" to a directory containing one of the above files.  If
  "glslang" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  cmake/HelperFunctions.cmake:262 (include)
  plugins/timewarp_gl/CMakeLists.txt:2 (get_external_for_plugin)
```

Reply:
I encountered this error, but after I installed glslang-dev, this error is gone. But then I am running into additional missing libraries: 1. xcb-glx0-dev 2. libx11-xcb-dev 3. libxcb-randr0-dev 4. xkbcommon-dev. Finally, I am reaching the error:
```
CMake Error at cmake/GetMonado_gl.cmake:20 (find_package):
  By not providing "FindOpenXR.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "OpenXR", but
  CMake did not find one.

  Could not find a package configuration file provided by "OpenXR" with any
  of the following names:

    OpenXRConfig.cmake
    openxr-config.cmake

  Add the installation prefix of "OpenXR" to CMAKE_PREFIX_PATH or set
  "OpenXR_DIR" to a directory containing one of the above files.  If "OpenXR"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  cmake/HelperFunctions.cmake:262 (include)
  plugins/timewarp_gl/CMakeLists.txt:2 (get_external_for_plugin)
```

Reply:
Unfortunately, neither glslang-dev nor glslang-tools are working for me. I'm inclined to add a FindGlslang.cmake file because the Vulkan config will also depend on it.

Reply:
@Jebbly What Operating systems and versions are you using?

Reply:
@jianxiapyh All of those libraries are listed as dependencies in the documentation as is libopenxr-dev which should provide OpenXRConfig.cmake

Reply:
@astro-friedel , I don't think libopenxr-dev is in 20.04

Reply:
@astro-friedel 
I found a solution to this OpenXRConfig. According to https://monado.freedesktop.org/packages-ubuntu.html
For 20.04, we need to add
```
sudo add-apt-repository ppa:monado-xr/monado
sudo apt-get update|
```
in order to install libopenxr-dev

Reply:
Yes, adding that repo has been part of the instructions for 20.04 for a while.

Reply:
> @Jebbly What Operating systems and versions are you using?

I'm on Ubuntu 20.04, and still running into this issue.

Reply:
Okay, I think I've finally resolved the issue. The issue has to do with installing the Vulkan SDK according to this documentation: https://vulkan.lunarg.com/doc/view/latest/linux/getting_started_ubuntu.html

When that additional repository is added, installing glslang causes an error for some reason (likely due to versioning, but not entirely sure). It would be good to document this behavior somewhere because it seems likely for someone to encounter this in the future.